internal/server/state.go

```diff
@@ -0,0 +1,136 @@
+/*
```

Comment:
Should we have a stateful server?

### Stateful Server:
- PROs:
  - User can pass fewer arguments -> more automation
- CONs:
  - Need to handle state -> prone to bugs
  
### Stateless server:
- PROs:
  - No need to handle state -> fewer chance to introduce bugs
- CONs:
  - State is 'handled' by the callee
  - User must provide lot more arguments

Reply:
I like automation so I would go for a stateful server, but I'd like other opinions on this matter

Reply:
cc @osinstom @pudelkoM ( also @daniele-moro even though we discussed about this and already know your opinion)

Reply:
I'm not really knowledgeable enough to make a call here, sorry.

Reply:
I'm inclined towards the stateless approach since this CLI will be used in many tests (UP4, TestON), I'd prefer to have more freedom in using it, rather than being bounded by the internal implementation or adding complexity inside it. Also, the stateless one seems way more similar to the python-based one, and thus, would require less work when we'll move to the go-based CLI in TestON (cc. @jhall11 ).

Reply:
A client (CLI/mock-smf/whatever) should definitely have a way to specify parameters such as UE Pool or eNB address. A way to achieve that would be to implement `Configure` RPC as you suggested, Emanuele. However, how does a user select eNB address to use for a given UE session? It should be definitely feasible to adjust that. Let's imagine we want to simulate a handover scenario - the stateful option would require extensions to `pfcpsimServer` to support that. On the other hand, the stateless option gives a client flexibility to just send a MODIFY with different FAR parameters. 

Taking above into consideration, **I'm also in favor of the stateless approach.**

Reply:
we still need to save the established sessions, since we need the remoteSEID that was used by the UPF, otherwise when trying to modify a session, if remoteSEID does not match with UPF's localSEID, the session modification's response has `Cause Rejected` and the session is not modified.

I thought that we could guess the remoteSEID but apparently it's a random so no way to guess it 

https://github.com/omec-project/upf-epc/blob/5fe53d0b92b88b451ab6c03c54741a55bef63771/pfcpiface/sessions.go#L40

So a fully stateless approach is not possible (even`mock-smf.py` is not fully stateless)

Reply:
Yes, I believe it's ok if it's not fully stateless. For the first implementation, we should aim at having a go-based mock-SMF as a drop-in replacement for the python-based one. Unless we see a huge room for improvement or the implementation becomes difficult/ugly and needs a rethinking, I would stick with the "stateless" approach (with the smallest possible amount of state kept in the mock SMF).